src/sql/workbench/services/notebook/browser/notebookServiceImpl.ts

```diff
@@ -678,7 +692,9 @@ export class NotebookService extends Disposable implements INotebookService {
 		let promises: Promise<ISerializationProvider | undefined>[] = [
 			providerDescriptor.instanceReady,
 			new Promise<ISerializationProvider | undefined>((resolve, reject) => setTimeout(() => {
-				onUnexpectedError(localize('serializationProviderTimeout', 'Waiting for Serialization Provider availability timed out for notebook provider \'{0}\'', providerDescriptor.providerId));
+				if (!providerDescriptor.instance) {
+					onUnexpectedError(localize('serializationProviderTimeout', 'Waiting for Serialization Provider availability timed out for notebook provider \'{0}\'', providerDescriptor.providerId));
```

Comment:
You could also instead just have this promise reject and then have the Promise.race call onUnexpectedError if an error occurs. That should accomplish the same thing and be more aligned with how Promise.race is supposed to be used.